astropy/units/quantity_helper/function_helpers.py

```diff
@@ -241,6 +241,10 @@ def like_helper(a, *args, **kwargs):
     unit = a.unit if subok else None
     return (a.view(np.ndarray),) + args, kwargs, unit, None
 
+@dispatched_function
+def full(shape, fill_value, dtype=None, order='C'):
+    return np.full(shape, fill_value, dtype, order), fill_value.unit, None
```

Comment:
This could also be something like:
```suggestion
    return np.full(shape, fill_value.view(np.ndarray), dtype, order), fill_value.unit, None
```

Reply:
You do not know here that `fill_value` is a `Quantity` - because you will have gotten here by the `like` argument. So, you'd need to pass it through `_as_quantity()`. I also think this works fine as a `function_helper`, i.e.,
```
@function_helper
def full(shape, fill_value, dtype=None, order='C'):
    fill_value = _as_quantity(fill_value)
    return (shape, fill_value.value, dtype, order), {}, fill_value.unit, None
```
```

With that, you can make it a `@function_helper` and

Reply:
Hmm, so something like `np.full(10, fill_value=1, like=my_quantity)` is expected behavior too?

Reply:
I think the idea is you pass in a class, as in `like=Quantity`, and for `Quantity`, `fill_value` does not have to have units. But you make a good point that for `Angle` there should be units, so my solution is not correct for that. Unfortunately, in this piece of code we do not actually know what (sub)class of Quantity we're in! We only know inside `__array_function__`... I think this will need some looking into how it actually is called...